### DISCUSSION TITLE
How to integrate that Application (by synchronizing "users" table of migration) to our existing Laravel App ???

### DISCUSSION THREAD

**Discussion by chegmarco1989** [RTippin/messenger#62] [Q&A]
Hello.

I have a serious headache. 
We already have a Laravel Application which has a Migration (Table) `users` whose content is:

```
public function up()
    {
        Schema::create('users', function (Blueprint $table) {
            $table->increments('id');
            $table->string('surname');
            $table->string('first_name');
            $table->string('last_name')->nullable();
            $table->string('username')->unique();
            $table->string('email')->nullable();
            $table->string('password');
            $table->char('language', 7)->default('en');
            $table->rememberToken();
            $table->softDeletes();
            $table->timestamps();
        });
    }
```

When that of **"Messenger-Demo" (https://github.com/RTippin/messenger-demo)** is:

```
public function up()
    {
        Schema::create('users', function (Blueprint $table) {
            $table->uuid('id')->primary();
            $table->string('name');
            $table->string('email')->unique();
            $table->string('avatar')->nullable();
            $table->boolean('admin')->default(false);
            $table->boolean('demo')->default(false);
            $table->string('password');
            $table->rememberToken();
            $table->timestamps();
        });
    }
```
We want to synchronize both Applications so that **Messenger-Demo** uses the same `users` Table (same DataBase) with our first Application that is running. 

So, we want to know how to do that ???
In other word, how to change `$table->uuid('id')->primary();` by `$table->increments('id');` And: `$table->string('name');` by these two `$table->string('surname');` and `$table->string('first_name');` without getting any error in Views ???

**_In which part of controllers, views or other code should we make modification if we change `users` migration to use the same `users` Table of our existing Laravel Application ???_**

Please, help us.

**Comment by RTippin** [comment]
Not sure I understand what you are trying to accomplish, as it sounds like you are wanting to run two backends at the same time under the same database. My demo app is not meant to be used in combination with your existing app, it is just one example of it being installed into a full laravel app, with config and assets published. If you have an existing app, then please follow the install docs for this `rtippin/messenger` :

> https://github.com/RTippin/messenger/blob/1.x/docs/Installation.md

In there it goes over the interface your user model implements and how to set the properties to overwrite name/avatar and other columns needed. Your user models ID can be UUID or int. If you need UUIDS, it can be configured during installation with the uuid flag, thus altering how the migrations are executed. `php artisan messenger:install --uuids`

Once the core is installed, you can then install the UI addon package:

> https://github.com/RTippin/messenger-ui/blob/master/README.md#installation

As well as the other optional packages for `rtippin/messenger`:

> https://github.com/RTippin/messenger-bots/blob/master/README.md#installation
> https://github.com/RTippin/messenger-faker/blob/master/README.md#installation

**Comment by chegmarco1989** [comment]
Hi @RTippin and thanks a lot for your feedback. 
I'm confuse.
How to access the Route of `messenger` after installed **core** and **UI addon package** from my web application ???

Does **this Route is automacally generated after installed** both **core** and **UI addon package** from my web application ???

If so, How to access it (that route) after installed them (core and pakages) from My Laravel existing Application ???

**Comment by chegmarco1989** [comment]
Hi over here.

Does anyone can help me or clear me ??? @RTippin ???

I'm confuse.
How to access the Route of messenger after installed core and UI addon package from my web application ???

Does this Route is automacally generated after installed both core and UI addon package from my web application ???

If so, How to access it (that route) after installed them (core and pakages) from My Laravel existing Application ???

I'm still waiting.

**Comment by RTippin** [comment]
@chegmarco1989 Have you tried to install the UI package following the links I've sent? I'd reccomend trying this stuff on your own. The docs in my UI package explain the config and paths to access the published UI.

Also, please be patient, it was my weekend, and I have a full time job, so my replies here will be limited.

**Comment by chegmarco1989** [comment]
Hi @RTippin and thanks for your feedback.

Yes We already install UI Package that is why I asked you waht is the next step to access on UI installed ???

Does this generate automacally Route because we don't see anything about "Route" on: https://github.com/RTippin/messenger-ui/blob/master/README.md#installation ???

How to access Messenger after instally all about it (UI Package included) ???

Please answer us.

**Comment by RTippin** [comment]
It is shown in the config section:

> https://github.com/RTippin/messenger-ui/blob/master/README.md#config

https://github.com/RTippin/messenger-ui/blob/aacfd10fce3b51cfb2b1dde1d1ee668603acbedd/config/messenger-ui.php#L43

By default, it would be `yoursite.com/messenger`

**Comment by chegmarco1989** [comment]
Thanks again @RTippin  for your response.

If We want to put `Messenger` (on SUBDOMAIN like this: `https://meet.ourdomain.name`), what will be the `domain` and its prefix ??? 
Does this looks right ???

```
'routing' => [
        'domain' => "meet.ourdomain.name",
        'prefix' => 'null',
        'middleware' => ['web', 'auth', 'messenger.provider'],
        'invite_middleware' => ['web', 'messenger.provider'],
    ],
```

is it correct ??? If not, please could you help us to do it ???
